src/libdrakvuf/libdrakvuf.h

```diff
@@ -274,6 +302,32 @@ char *drakvuf_get_current_process_name(drakvuf_t drakvuf,
                                        uint64_t vcpu_id,
                                        x86_registers_t *regs);
 
+
+bool drakvuf_get_current_thread_id( drakvuf_t drakvuf, 
+                                    uint64_t vcpu_id, 
+                                    x86_registers_t *regs,
+                                    uint32_t *thread_id );
+
+bool drakvuf_get_previous_mode( drakvuf_t drakvuf, 
```

Comment:
[This link from Microsoft](https://msdn.microsoft.com/en-us/library/windows/hardware/ff559860(v=vs.85).aspx) explains the PreviousMode field.

It's very useful when using within the hooks for the arguments treatment (that's the reason why it uses the current thread address)